File: php/testcode/pathtraver_006.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_pt_unlink_basename
function pathtraver_unlink_basename(BenchmarkRequest $req): BenchmarkResponse {
    $filename = $req->param('filename');
    $safe = basename($filename);
    unlink("/tmp/" . $safe); // vuln-code-snippet safe-line php_pt_unlink_basename
    return BenchmarkResponse::ok("File deleted");
}
// vuln-code-snippet end php_pt_unlink_basename

File: ruby/testcode/pathtraver_006.rb
require_relative 'shared'

# vuln-code-snippet start ruby_pt_join_basename
def serve_asset(req)
  base_dir = "/assets/files"
  filename = req.param('filename')
  full_path = File.join(base_dir, File.basename(filename)) # vuln-code-snippet safe-line ruby_pt_join_basename
  content = File.read(full_path)
  BenchmarkResponse.ok(content)
end
# vuln-code-snippet end ruby_pt_join_basename
